packages/blob/src/index.ts

```diff
@@ -122,6 +116,96 @@ export async function put(
   return blobResult;
 }
 
+// vercelBlob.clientPut()
+// This is a wrapper that will fetch the client token for you and then upload the file
+export interface ClientPutCommandOptions extends BlobCommandOptions {
+  access: 'public';
+  contentType?: string;
+  handleClientUploadUrl: string;
+}
+
+export async function clientPut(
+  pathname: string,
+  body:
+    | string
+    | Readable
+    | Blob
+    | ArrayBuffer
+    | FormData
+    | ReadableStream
+    | File,
+  options: ClientPutCommandOptions
+): Promise<PutBlobResult> {
+  if (!pathname) {
```

Comment:
should we throw if `typeof window === undefined`? not 100% sure that the request comes from the client, but it could still be useful

Reply:
Same as https://github.com/vercel/storage/pull/347/files#r1313037483, we should not artificially limit where put() can be used only because we provide a convenience clientPut() method. Some people will not want to use it.